common/workunit/workunit.cpp

```diff
@@ -6805,6 +6863,140 @@ const char *clusterTypeString(ClusterType clusterType, bool lcrSensitive)
     throwUnexpected();
 }
 
+bool extractFromWorkunitDAToken(const char * distributedAccessToken, StringBuffer * wuid, StringBuffer * user, StringBuffer * privKey)
+{
+    if (strncmp(distributedAccessToken, "HPCC[", 5))
+    {
+        DBGLOG("Not a valid workunit distributed access token");
+        return false;
+    }
+
+    const char * finger;
+    if (user)
+    {
+        finger = strstr(distributedAccessToken+5, "u=");
```

Comment:
what if another name ends with u=... for example au=123,fu=456,u=john?

Reply:
The only ones are u=xxx and w=yyy